api/tests/unit/organisations/test_unit_organisations_tasks.py

```diff
@@ -222,3 +224,84 @@ def test_send_org_subscription_cancelled_alert(db: None, mocker: MockerFixture)
         recipient_list=[],
         fail_silently=True,
     )
+
+
+def test_handle_api_usage_notifications(
+    mocker: MockerFixture,
+    organisation: Organisation,
+) -> None:
+    # Given
+    now = timezone.now()
+    OrganisationSubscriptionInformationCache.objects.create(
+        organisation=organisation,
+        allowed_seats=10,
+        allowed_projects=3,
+        allowed_30d_api_calls=100,
+        chargebee_email="test@example.com",
+        current_billing_term_starts_at=now - timedelta(days=45),
+        current_billing_term_ends_at=now + timedelta(days=320),
+    )
+    mock_api_usage = mocker.patch(
+        "organisations.tasks.get_current_api_usage",
+    )
+    mock_api_usage.return_value = 91
+    mock_send_mail = mocker.patch(
+        "organisations.tasks.send_mail",
+    )
+    assert not OranisationAPIUsageNotification.objects.filter(
+        organisation=organisation,
+    ).exists()
+
+    # When
+    handle_api_usage_notifications()
+
+    # Then
+    mock_api_usage.assert_called_once_with(organisation.id, "14d")
+    mock_send_mail.assert_called_once_with(
+        subject="Flagsmith API use has reached 90%",
+        message=(
+            "Hi there,\n\nThe API usage for Test Org has reached "
+            "90% within the current subscription period. Please "
+            "consider upgrading your organisations account limits.\n\n"
+            "Thank you!\n\nThe Flagsmith Team\n"
+        ),
+        html_message=(
+            "<table>\n\n        <tr>\n\n               "
+            "<td>Hi there,</td>\n\n        </tr>\n\n        "
+            "<tr>\n\n               <td>\n                 "
+            "The API usage for Test Org has reached\n                 "
+            "90% within the current subscription period.\n                 "
+            "Please consider upgrading your organisations account limits.\n"
+            "               </td>\n\n\n        </tr>\n\n        "
+            "<tr>\n\n               <td>Thank you!</td>\n\n      "
+            "  </tr>\n\n        <tr>\n\n               "
+            "<td>The Flagsmith Team</td>\n\n        "
+            "</tr>\n\n</table>\n"
+        ),
+        from_email="noreply@flagsmith.com",
+        recipient_list=["admin@example.com"],
+        fail_silently=True,
+    )
```

Comment:
I guess there's an argument to be said that we should parametrise this test to handle the different threshold 'groups'. The logic is pretty basic so I'll leave it for the purposes of approving this PR, but I think strictly we should test both paths.

Reply:
Sounds good. I'm going to write a separate test though. The output of the blobs of parameterization look ugly when the test suite is run with them, see below:

<img width="1511" alt="image" src="https://github.com/Flagsmith/flagsmith/assets/222715/84fae9e7-27a3-419f-939b-2de36ba8688c">